src/coordinator/ts/m3db/block/iterator_test.go

```diff
@@ -0,0 +1,336 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package block
+
+import (
+	"math"
+	"testing"
+	"time"
+
+	"github.com/m3db/m3db/src/coordinator/block"
+	coordtest "github.com/m3db/m3db/src/coordinator/test"
+	"github.com/m3db/m3db/src/dbnode/encoding"
+	"github.com/m3db/m3db/src/dbnode/ts"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	now = time.Now()
+	nan = math.NaN()
+)
+
+func createDatapoints(t *testing.T, timeInSeconds []int, vals []float64, now time.Time) []ts.Datapoint {
+	if len(timeInSeconds) != len(vals) {
+		t.Fatal("length of values must equal length of timestamps")
+	}
+
+	dps := make([]ts.Datapoint, len(vals))
+	for i, val := range vals {
+		dps[i] = ts.Datapoint{
+			Timestamp: now.Add(time.Duration(timeInSeconds[i]) * time.Second),
+			Value:     val,
+		}
+	}
+
+	return dps
+}
+
+func createNSBlockIter(iters []encoding.SeriesIterator, start, end time.Time, stepSize time.Duration) consolidatedNSBlockIter {
+	return consolidatedNSBlockIter{
+		consolidatedNSBlockSeriesIters: iters,
+		bounds: block.Bounds{
+			Start:    start,
+			End:      end,
+			StepSize: stepSize,
+		},
+		indexTime: start.Add(-1 * stepSize),
+	}
+}
+
+type testCase struct {
+	start, end                     time.Time
+	stepSize                       time.Duration
+	dps                            [][]ts.Datapoint
+	expectedResults, actualResults []float64
```

Comment:
nit: would prefer to have actualResults be a variable built in the test, rather than added to the struct itself